lib/reply.js

```diff
@@ -148,14 +148,22 @@ Reply.prototype.send = function (payload) {
       return this
     }
 
-    if (Buffer.isBuffer(payload)) {
+    if (payload instanceof Uint8Array) {
       if (hasContentType === false) {
         this[kReplyHeaders]['content-type'] = CONTENT_TYPE.OCTET
       }
       onSendHook(this, payload)
       return this
     }
 
+    if (payload.buffer instanceof ArrayBuffer) {
```

Comment:
I think we can avoid this duplication by editing the previous code:

```
const isBuffer = Buffer.isBuffer(payload)
const isTypedArray = payload?.buffer instanceof ArrayBuffer
const payloadToSubmit = isTypedArray ? payload?.buffer : payload
```

Reply:
If `payload` is an instance of `Buffer` , `payload?.buffer instanceof ArrayBuffer` will return true even if it is not a TypedArray because `Buffer` extends `Uint8Array`

Reply:
wdyt about the following?

```js
const isBuffer = Buffer.isBuffer(payload);
const isTypedArray = !isBuffer && payload?.buffer instanceof ArrayBuffer;
const isBufferLike = isBuffer || isTypedArray;
```

This way, we can replace the two checks with a single one, where we initially check if the payload is a _buffer-like_ payload.
If `true`, we enter the `if` and attach the headers, and we use the previous check to decide how to send the payload.

Reply:
I was thinking something like this:
```js
if (payload?.buffer instanceof ArrayBuffer) {
    if (hasContentType === false) {
        this[kReplyHeaders]['content-type'] = CONTENT_TYPE.OCTET
    }
    const payloadToSend = payload instanceof Uint8Array ? payload : Buffer.from(payload.buffer)
    onSendHook(this, payloadToSend)
    return this
}
```
wdyt?

Reply:
Ah, right, forgot `Buffer` contains the `buffer` prop with an `ArrayBuffer` instance underneath. Then, yeah, LGTM 👍.

Just a note:

To make it safer, might be better to go with `Buffer.isBuffer` to look for the `Buffer` instance, as if `Uint8Array` is passed, it will be truncated as explained earlier. i.e.

```js
const payloadToSend = Buffer.isBuffer(payload) ? payload : Buffer.from(payload.buffer)
```